### PR TITLE
drop data variables not common to all datasets before combining them

### PR DESCRIPTION
**Pull request by mjlosch**
This fixes the problem described in #303: if in a series of output files from the diagnostics package some files contain more fields than others. Currently the `xr.combine_by_coords(..., combat='override', ...)` sets all fields to NaNs in files that contain fewer fields. 

Clearly, changing the number of fields in a diagnostics-output stream is not ideal, but this fix would have saved me from debugging my notebooks for a few days.

### CODE REVIEW

File: xmitgcm/mds_store.py [hunk 1]
```diff
@@ -239,6 +239,20 @@ def open_mdsdataset(data_dir, grid_dir=None,
                 datasets = [open_mdsdataset(
                         data_dir, iters=iternum, read_grid=False, **kwargs)
                     for iternum in iters]
+                # drop all data variables not common to the datasets,
+                # this should be done by xr.combine_by_coords, but
+                # with the "overide" option, it does not work properly
+                all_data_vars = set(datasets[0].data_vars)
```
Comment (timothyas) [inline xmitgcm/mds_store.py:245]: This line will fail if `datasets == []`, which can happen and then the user will probably be staring at a weird error.

**Comment by timothyas** [comment]
Hey @mjlosch, this looks generally good to me, but I made a detailed comment above. Since I mentioned that I would help add a test, I've made fixed that and added a test - would you like me to push those changes? The push would also contain a merge with upstream master, so we should actually see the tests pass.

File: xmitgcm/mds_store.py (hunk 1 above)
Comment (mjlosch) [reply xmitgcm/mds_store.py:245]: thanks for the comment. We could catch this case and only do lines 242 to 255 `if datasets != []`

Comment (timothyas) [reply xmitgcm/mds_store.py:245]: That is what I was thinking as well. Do you want me to push those changes to your branch?

Comment (mjlosch) [reply xmitgcm/mds_store.py:245]: Yes, please, go ahead

**Comment by timothyas** [comment]
Ah, good to see the green check mark. I made a test that copies the LLC90 meta/data pair to a new iteration and hides "ETAN" from the new meta file. I originally made this as the last test in the `test_mds_store.py` module, and when I did this, the following would pass: `pytest test_mds_store -k test_drop_uncommon_diagnostics`  but running the whole test module `pytest test_mds_store` would fail. I have no idea why, but by moving this test earlier in the sequence, everything works just fine.

**Comment by mjlosch** [comment]
looks great! At one point you'll have to explain to me the subtleties of using brackets or not.

I would have included all of the new code in the new if statement, but this looks very good, too.

Just a note: when I run the test locally I get a lot of "DeprecationWarning"s, but the do not have anything to do with the modifications of the PR.

I agree, it's ready to go.

**Comment by timothyas** [comment]
Great, thanks @mjlosch. I think I used brackets just to get a one liner, but it's not really necessary. I'm seeing the deprecation warnings too, it's on the todo list for the next plumbing update.

I think this has little chance of breaking anyone else's code, so unless there's any rebuttal I will merge this soon to push it through.

**Pull request merged**